File: lib/Widget/DrawerWidget/SplashScreen.java
import 'package:flutter/material.dart';
import 'dart:async';
import 'package:firebase_auth/firebase_auth.dart';
import '../../Conist.dart';
import '../Login&Sign/LoginPage.dart';
import '../Main/bottom_nav.dart'; // Make sure this import is correct for your BottomNav screen
import '../DoctorReservation/ReservationScreen.dart'; // Import for doctor home screen

class SplashScreen extends StatefulWidget {
  @override
  _SplashScreenState createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> {
  @override
  void initState() {
    super.initState();
    _checkAuthStatus();
  }

  Future<void> _checkAuthStatus() async {
    // Add a small delay for the splash screen
    await Future.delayed(Duration(seconds: 2));

    // Get the current user from Firebase Auth
    User? user = FirebaseAuth.instance.currentUser;

    if (user != null) {
      // User is logged in - check their role
      await _redirectBasedOnUserRole(user.uid);
    } else {
      // No user logged in - go to login screen
      Navigator.of(context).pushReplacement(
        MaterialPageRoute(builder: (context) => Login()),
      );
    }
  }

  Future<void> _redirectBasedOnUserRole(String userId) async {
    try {
      // Get user data from Firestore
      final userDoc = await FirebaseFirestore.instance
          .collection('users')
          .doc(userId)
          .get();

      if (userDoc.exists) {
        final role = userDoc.data()?['role'] ?? 'user';
        
        if (role == 'doctor') {
          Navigator.of(context).pushReplacement(
            MaterialPageRoute(builder: (context) => DoctorReservationPage()),
          );
        } else {
          Navigator.of(context).pushReplacement(
            MaterialPageRoute(builder: (context) => BottomNav()),
          );
        }
      } else {
        // User document doesn't exist - go to login
        Navigator.of(context).pushReplacement(
          MaterialPageRoute(builder: (context) => Login()),
        );
      }
    } catch (e) {
      // If any error occurs, redirect to login screen
      Navigator.of(context).pushReplacement(
        MaterialPageRoute(builder: (context) => Login()),
      );
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: BackgroundColor,
      body: Center(
        child: Image.asset(
          'assets/images/22.png',
          height: 200,
          width: 200,
        ),
      ),
    );
  }
}

File: lib/screens/SplashScreen/SplashScreen.dart
import 'package:flutter/material.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import '../../Conist.dart';
import '../../Widget/HomeWidget/bottom_nav.dart';
import '../Home/Doctor_Button.dart';
import '../Login&Sign/LoginPage.dart';

class SplashScreen extends StatefulWidget {
  @override
  _SplashScreenState createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> with TickerProviderStateMixin {
  late AnimationController _controller;
  late AnimationController _pulseController;
  late Animation<double> _logoScale;
  late Animation<double> _logoOpacity;
  late Animation<double> _pulseAnimation;
  late Animation<Offset> _textSlide;

  @override
  void initState() {
    super.initState();

    _controller = AnimationController(
      vsync: this,
      duration: Duration(milliseconds: 1500),
    );

    _pulseController = AnimationController(
      vsync: this,
      duration: Duration(milliseconds: 800),
    )..repeat(reverse: true);

    _logoScale = CurvedAnimation(
      parent: _controller,
      curve: Curves.elasticOut,
    ).drive(Tween<double>(begin: 0.7, end: 1.0));

    _logoOpacity = CurvedAnimation(
      parent: _controller,
      curve: Curves.easeIn,
    ).drive(Tween<double>(begin: 0.0, end: 1.0));

    _pulseAnimation = Tween<double>(begin: 0.8, end: 1.2).animate(
      CurvedAnimation(
        parent: _pulseController,
        curve: Curves.easeInOut,
      ),
    );

    _textSlide = Tween<Offset>(
      begin: Offset(0, 0.5),
      end: Offset.zero,
    ).animate(CurvedAnimation(
      parent: _controller,
      curve: Curves.easeOutQuad,
    ));

    _controller.forward();
    _initializeApp();
  }

  Future<void> _initializeApp() async {
    await Future.delayed(Duration(seconds: 2));

    try {
      User? user = FirebaseAuth.instance.currentUser;

      if (user == null) {
        _navigateToLogin();
        return;
      }

      final userDoc = await FirebaseFirestore.instance
          .collection('users')
          .doc(user.uid)
          .get();

      if (!userDoc.exists) {
        await FirebaseAuth.instance.signOut();
        _navigateToLogin();
        return;
      }

      final role = userDoc.data()?['role'] ?? 'user';
      await Future.delayed(Duration(milliseconds: 500));
      _navigateToHome(role);

    } catch (e) {
      _navigateToLogin();
    }
  }

  void _navigateToLogin() {
    Navigator.of(context).pushAndRemoveUntil(
      MaterialPageRoute(builder: (context) => Login()),
          (Route<dynamic> route) => false,
    );
  }

  void _navigateToHome(String role) {
    Widget homeScreen = role == 'doctor' ? DoctorDashboard() : BottomNav();
    Navigator.of(context).pushAndRemoveUntil(
      MaterialPageRoute(builder: (context) => homeScreen),
          (Route<dynamic> route) => false,
    );
  }

  @override
  void dispose() {
    _controller.dispose();
    _pulseController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: BackgroundColor,
      body: Stack(
        children: [
          Positioned.fill(
            child: CustomPaint(
              painter: _MedicalPatternPainter(),
            ),
          ),

          Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                ScaleTransition(
                  scale: _logoScale,
                  child: FadeTransition(
                    opacity: _logoOpacity,
                    child: Container(
                      padding: EdgeInsets.all(24),
                      decoration: BoxDecoration(
                        color: Colors.white,
                        shape: BoxShape.circle,
                        boxShadow: [
                          BoxShadow(
                            color: Colors.black.withOpacity(0.1),
                            blurRadius: 15,
                            spreadRadius: 2,
                          )
                        ],
                      ),
                      child: Image.asset(
                        'assets/images/22.png',
                        height: 120,
                        width: 120,
                      ),
                    ),
                  ),
                ),

                SizedBox(height: 40),

                SlideTransition(
                  position: _textSlide,
                  child: Text(
                    'دليل السكر',
                    style: TextStyle(
                      fontSize: 32,
                      fontWeight: FontWeight.bold,
                      color: PrimaryColor,
                      letterSpacing: 1.2,
                    ),
                  ),
                ),

                SizedBox(height: 8),

                SlideTransition(
                  position: _textSlide,
                  child: Text(
                    'شريكك الطبي الموثوق',
                    style: TextStyle(
                      fontSize: 16,
                      color: SecondryColor,
                      fontStyle: FontStyle.italic,
                    ),
                  ),
                ),

                SizedBox(height: 40),

                ScaleTransition(
                  scale: _pulseAnimation,
                  child: Container(
                    width: 12,
                    height: 12,
                    decoration: BoxDecoration(
                      color: PrimaryColor,
                      shape: BoxShape.circle,
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _MedicalPatternPainter extends CustomPainter {
  @override
  void paint(Canvas canvas, Size size) {
    final paint = Paint()
      ..color = PrimaryColor.withOpacity(0.05)
      ..style = PaintingStyle.stroke
      ..strokeWidth = 1.5;

    final crossSize = 40.0;
    final spacing = 60.0;

    for (double x = 0; x < size.width; x += spacing) {
      for (double y = 0; y < size.height; y += spacing) {
        canvas.drawLine(
          Offset(x - crossSize / 2, y),
          Offset(x + crossSize / 2, y),
          paint,
        );
        canvas.drawLine(
          Offset(x, y - crossSize / 2),
          Offset(x, y + crossSize / 2),
          paint,
        );
      }
    }
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) => false;
}
